### PR TITLE
Applications routing feature

### PR DESCRIPTION
**Pull request by tahmidefaz**

### CODE REVIEW

**Comment by tahmidefaz** [comment]
@ryelo @karelhala ... this currently has a bug and I am not sure about. 
Like `https://ci.foo.redhat.com:1337/settings/general/` takes you to `https://ci.foo.redhat.com:1337/settings/general/general`.  Also, `https://ci.foo.redhat.com:1337/settings/applications/appName` doesn't route me to the `Applications` component, however `https://ci.foo.redhat.com:1337/settings/general/applications/appName` does. Any idea what I am doing wrong here?

File: src/Routes.js [hunk 1]
```diff
@@ -29,6 +32,7 @@ export const Routes = (props) => {
     return (
         <Switch>
             <InsightsRoute path={ paths.general } component={ General } rootClass='general'/>
+            <InsightsRoute exact path={ paths.applications } component={ Applications } rootClass='applications'/>
 
             { /* Finally, catch all unmatched routes */ }
             <Route render={ () => some(paths, p => p === path) ? null : (<Redirect to={ paths.general }/>) }/>
```
Comment (tahmidefaz) [inline src/Routes.js]: My best guess is that this is what's causing the aforementioned issue, but I don't have any clue...

Comment (karelhala) [reply src/Routes.js]: I think the problem is with this function https://github.com/RedHatInsights/frontend-components/blob/master/packages/utils/src/helpers.js#L70

I'll write a quick fix for it.

Comment (tahmidefaz) [reply src/Routes.js]: Oh, great! Thanks a lot!

Comment (karelhala) [reply src/Routes.js]: This should do the trick https://github.com/RedHatInsights/frontend-components/pull/265

Comment (tahmidefaz) [reply src/Routes.js]: Thanks a lot! Works like a charm! :star2:

**Comment by tahmidefaz** [comment]
Looks like @karelhala 's magic fixed it! :stars: 
Any improvements needed to be made here? 
Jira: https://projects.engineering.redhat.com/browse/RHCLOUD-2973

**Comment by tahmidefaz** [comment]
Just added some additional routing options within `Applications.js`. We may or may not need it...

File: src/SmartComponents/Applications/Applications.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        const TestApp = asyncComponent(() => import(/* webpackChunkName: "TestApp" */ './TestApp'));
+        return (
+            <React.Fragment>
+                <PageHeader>
+                    <PageHeaderTitle title='Applications Settings'/>
+                    <p>{ `Settings for ${ appName }` }</p>
+                </PageHeader>
+                <Switch>
+                    <Route exact path='/applications/testApp' component={ TestApp }/>
```
Comment (karelhala) [inline src/SmartComponents/Applications/Applications.js]: This `Route` is not really needed. We'll pull the settings from server and display it here without any nested routes.

Comment (tahmidefaz) [reply src/SmartComponents/Applications/Applications.js]: Sound good to me!

**Pull request merged**